accepted/future-releases/abstract-external-fields/implementation-plan.md

```diff
@@ -0,0 +1,153 @@
+# Implementation Plan for "External and Abstract Variables"
+
+Owner: eernst@google.com ([@eernstg](https://github.com/lrhn/) on GitHub)
+
+Relevant links:
+
+* [Tracking issue](https://github.com/dart-lang/sdk/issues/42560)
+* [Feature specification](https://github.com/dart-lang/language/blob/master/accepted/future-releases/abstract-external-fields/feature-specification.md)
+
+## Phase 0 (Preliminary steps)
+
+### "external-abstract-variables" Experimental flag
+
+The implementation of this feature is hidden behind an
+[experiment flag][].
+The flag `--enable-experiment=external-abstract-variables`
```

Comment:
Is there a connection between the non-nullable flag and this flag (will one imply the other)?

dart:ffi uses external fields to support nnbd so they needs together.

Reply:
Yes, the need gets acute with null-safety, and it's much more like a convenience for legacy code. For instance, `int x;` is an error with null-safety, so there's a need to initialize `x` (even though that's just useless noise, as seen from the perspective that `x` only exists in order to induce a getter and a setter).

So the intention is that this feature only exists when null-safety is enabled. I assume it's helpful to make this strict (such that there's no need to support the feature at all in legacy mode), so I've added that to the document.

Reply:
The current CFE support for external fields is guarded by `--enable-experiment=non-nullable`.

Reply:
I think we should consider just keeping this under the non-nullable flag and committing to ship it as part of the null safety release given the current timing.  cc @mit-mit @devoncarew for thoughts on this.

Reply:
I have no concerns with that -

Reply:
It sounds like we're leaning toward reusing `--enable-experiment=non-nullable` for this feature, to ensure that it is inseparable from null-safety.

Reply:
Changed the text to use 'non-nullable'.